browser/src/UI/components/Tabs.tsx

```diff
@@ -186,7 +186,7 @@ const mapStateToProps = (state: State.IState, ownProps: ITabContainerProps): ITa
     const shouldUseVimTabs = state.configuration["tabs.showVimTabs"]
     const tabs = shouldUseVimTabs ? getTabsFromVimTabs(state) : getTabsFromBuffers(state)
 
-    const visible = state.configuration["tabs.enabled"]
+    const visible = state.configuration["tabs.enabled"] && shouldUseVimTabs !== "native"
```

Comment:
I feel this is making configuration very complex with settings that depend on context of other settings. It's good for retrocompatibility but bad in the long run.

What about: `tabs.showVimTabs = (buffers)|native|files`

Reply:
Yep, that seems fair. Perhaps a `tabs.tabType` option instead for buffers/tabs/native vim tabs?

Reply:
Well, `tabs.enabled` seems like a different configuration altogether, not sure it could be replaces.

Thoughts @bryphe ?

Reply:
Thinking about this some more since the comment on #992 , perhaps we can streamline it to not require a new setting.

Here's a table of the scenarios we're trying to support:

| Scenario | Description | `tabs.enabled` | `tabs.showVimTabs` |
| --- | --- | --- | --- |
| Default tab experience | Show buffers in Oni-style UX | `true` (default) | `false` (default) |
| Disabled | No tabs shown at all | `false`  | `false` (default) |
| Vim tabs in Oni | Show the vim tabs, in Oni-style UX | `true` (default) | `true` |
| Vim tabs, native UX | Show the vim tabs, w/o Oni UX | `false` | `true` |

The 'happy-path' IMO, is using the default experience, or turning it off - that means the 'happy-path' is completely controllable via `tabs.enabled`. I want to make sure those 'happy-path' options are very simple and streamlined to configure.

However, the more 'advanced path' (using Vim tabs with Oni's UX, or falling back to the native Vim UX entirely), also requires toggling  `tabs.showVimTabs`, with the most advanced configuration requiring changing both `tabs.enabled` to `true` and `tabs.showVimTabs` to `false` (enabling the native vim UX). I think that is pretty reasonable, as long as we document it. 

That would simplify the configuration setting and still support the ability to 'eject' and use the default VIM tab UX - does that seem reasonable?

Reply:
I think the idea is fine, the only problem is that to enable a specific configuration I want I'll have to check a 'truth table'.

I think it would be much more straightforward to enable or disable tabs and then (if they are enabled) choose explicitly the behaviour as a string (buffers|tabs|native).

This is just my aversion to configurations that depend on each other.

Reply:
> the only problem is that to enable a specific configuration I want I'll have to check a 'truth table'.

Fair criticism 😄 

Perhaps this is clearer:
`tabs.enabled`: behaves as-is today
`tabs.mode`: `'buffers'|'tabs'|'native'`

It's basically your suggestion above - just renaming `tabs.showVimTabs` to `tabs.mode`. I realized the only real issue I had with the specifying the behavior directly was being confused by the case `tabs.showVimTabs`: `'buffers'` - but if we rename the setting, it alleviates my concern. Only downside is changing the setting would be a breaking change.

Reply:
> Only downside is changing the setting would be a breaking change.

That's the advantage of being a 0.2.x version :-). We need to make it right now not to have to deal with this in the future